src/archivematicaCommon/lib/databaseFunctions.py

```diff
@@ -229,8 +229,12 @@ def logTaskCompletedSQL(task):
     print("Logging task output to db", task.UUID)
     taskUUID = task.UUID.__str__()
     exitCode = task.results["exitCode"].__str__()
-    stdOut = task.results["stdOut"]
-    stdError = task.results["stdError"]
+    # truncate task stdout/stderror logged to the database
+    # (to prevent mysql server errors, refs. #10732) 
+    TASK_STDOUT_LIMIT = 16*1024*1024
+    TASK_STDERR_LIMIT = 16*1024*1024
+    stdOut = (task.results["stdOut"][:TASK_STDOUT_LIMIT] + '..') if len(task.results["stdOut"]) > TASK_STDOUT_LIMIT else task.results["stdOut"]
+    stdError = (task.results["stdError"][:TASK_STDERR_LIMIT] + '..') if len(task.results["stdError"]) > TASK_STDERR_LIMIT else task.results["stdError"]
```

Comment:
Do you know if the types of `task.results["stdError"]` and `task.results["stdOut"]` are always going to be Py2 `str` (aka `bytes`)? If that's the case I think we'll be fine but if you expect `unicode` strings then we should [encode](https://github.com/artefactual/archivematica/blob/stable/1.5.x/src/archivematicaCommon/lib/archivematicaFunctions.py#L59-L62) them before so we can effectively see its size in bytes. @jrwdunham may know more about this, he's dealt with similar stuff before.